zqd/handlers_test.go

```diff
@@ -47,6 +47,41 @@ func TestSearch(t *testing.T) {
 	require.Equal(t, test.Trim(src), res)
 }
 
+func TestSearchNoCtrl(t *testing.T) {
+	src := `
+#0:record[_path:string,ts:time,uid:bstring]
+0:[conn;1521911723.205187;CBrzd94qfowOqJwCHa;]
+0:[conn;1521911721.255387;C8Tful1TvM3Zf5x8fl;]
+`
+	_, client, done := newCore(t)
+	defer done()
+	sp, err := client.SpacePost(context.Background(), api.SpacePostRequest{Name: "test"})
+	require.NoError(t, err)
+	_ = postSpaceLogs(t, client, sp.Name, nil, false, src)
+
+	parsed, err := zql.ParseProc("*")
+	require.NoError(t, err)
+	proc, err := json.Marshal(parsed)
+	require.NoError(t, err)
+	req := api.SearchRequest{
+		Space: "test",
+		Proc:  proc,
+		Span:  nano.MaxSpan,
+		Dir:   -1,
+	}
+	r, err := client.Search(context.Background(), req, map[string]string{"noctrl": "true"})
```

Comment:
I was confused about how this was sending back zng without a format query param; the client always specifies zng, we'll have to change that in https://github.com/brimsec/zq/issues/744 .

Reply:
👍 agree.